packages/next-codemod/transforms/lib/async-request-api/utils.ts

```diff
@@ -111,46 +102,25 @@ export function isMatchedFunctionExported(
   return isNamedExport
 }
 
-export function determineClientDirective(
-  root: Collection<any>,
-  j: JSCodeshift,
-  source: string
-) {
-  const hasStringDirective =
-    root
-      .find(j.Literal)
-      .filter((path) => {
-        const expr = path.node
-
-        return (
-          expr.value === 'use client' && path.parentPath.node.type === 'Program'
-        )
-      })
-      .size() > 0
-
-  // 'use client';
-  const hasStringDirectiveWithSemicolon =
-    root
-      .find(j.StringLiteral)
-      .filter((path) => {
-        const expr = path.node
-        return (
-          expr.type === 'StringLiteral' &&
-          expr.value === 'use client' &&
-          path.parentPath.node.type === 'Program'
-        )
-      })
-      .size() > 0
-
-  if (hasStringDirective || hasStringDirectiveWithSemicolon) return true
-
-  // Since the client detection is not reliable with AST in jscodeshift,
-  // determine if 'use client' or "use client" is leading in the source code.
-  const trimmedSource = source.trim()
-  const containsClientDirective =
-    /^'use client'/.test(trimmedSource) || /^"use client"/g.test(trimmedSource)
+// directive is not parsed into AST, so we need to manually find it
+// by going through the tokens. Use the 1st string token as the directive
+export function determineClientDirective(root: Collection<any>) {
```

Comment:
This would be so much easier with Babel where we have a dedicate `directives` field. Odd that we have `j.Directive` yet `j(source).find(j.Directive)` doesn't work

Reply:
We can go through expression statements instead. Those nodes have the `directive` property still intact. The statements that have a directive and are top level, will be relevant.

Example: https://astexplorer.net/#/gist/c8c9d85f02158271f82e0dd033888a9e/01dcea732acf28418bc718c1077466aff4143735

But if that doesn't work in our setup, don't waste time trying to make it work. The false-positives are probably a code-smell anyway since they would also pollute grep results for `'use client'`.

Reply:
searched from program.directives now